core/contracts/DesignatedVoting.sol

```diff
@@ -27,9 +27,9 @@ contract DesignatedVoting is MultiRole, Withdrawable {
     // this contract.
     Finder private finder;
 
-    constructor(address finderAddress) public {
-        _createExclusiveRole(uint(Roles.Owner), uint(Roles.Owner), msg.sender);
-        _createExclusiveRole(uint(Roles.Voter), uint(Roles.Owner), msg.sender);
+    constructor(address finderAddress, address ownerAddress, address voterAddress) public {
+        _createExclusiveRole(uint(Roles.Owner), uint(Roles.Owner), ownerAddress);
+        _createExclusiveRole(uint(Roles.Voter), uint(Roles.Owner), voterAddress);
```

Comment:
Let's say the voter key is stolen. If the owner key is used to change the voter address in this contract, there's no way to add this new mapping to the factory contract, right?

Correct me if I'm wrong, but it seems like in the current circumstances, it would rarely (if ever) make sense for the owner to change the voterAddress. I feel like it might make sense to do one of the following:
- Make the voterAddress hardcoded/unchangeable in the DesignatedVoting contract. This would mean that the way to rectify a stolen voter key would be to just withdraw the funds from the contract and deploy a new contract.
- Have a special method on the Factory contract that allows an address to manually add an associated DesignatedVoting contract for itself.

What do you think?

Reply:
Implemented the second one.